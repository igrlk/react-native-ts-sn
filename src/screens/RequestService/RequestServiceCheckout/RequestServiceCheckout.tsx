import React from 'react';
import { View, Text, TouchableOpacity, KeyboardAvoidingView } from 'react-native';

import Screen from 'library/common/commonComponents/Screen';
import GoBackButton from 'library/common/commonComponents/Buttons/GoBackButton';
import Input from 'library/common/commonComponents/Inputs/Input';
import Button from 'library/common/commonComponents/Buttons/Button';
import RequestServiceInfo from 'resources/images/svg/RequestServiceInfo';
import { getNavigateTo } from 'library/utilities/navigator';
import useRequestServiceCheckout from './requestServiceCheckoutHooks/useRequestServiceCheckout';

import styles from './requestServiceCheckoutStyles';

export default function RequestServiceCheckout() {
	const {
		fullName,
		setFullName,
		address,
		setAddress,
		city,
		setCity,
		state,
		setState,
		zipCode,
		setZipCode,
		availableSamplesCount,
		selectedSamplesCount,
		decrementSelectedSamples,
		incrementSelectedSamples,
		makeCheckout,
		fullNameError,
		addressError,
		cityError,
		stateError,
		zipCodeError,
	} = useRequestServiceCheckout();

	return (
		<Screen>
			<KeyboardAvoidingView behavior='padding'>
				<GoBackButton>Request Service</GoBackButton>
				<View style={styles.container}>
					<View>
						<Text style={styles.title}>Number of samples</Text>
						<View style={styles.info}>
							<TouchableOpacity onPress={getNavigateTo('FAQ', { toSamples: true })}>
								<RequestServiceInfo />
							</TouchableOpacity>
							<Text style={styles.infoText}>$300/ sample</Text>
						</View>
					</View>

					<View style={styles.controls}>
						<TouchableOpacity style={styles.control} onPress={incrementSelectedSamples}>
							<Text
								style={[
									styles.controlText,
									selectedSamplesCount === availableSamplesCount && styles.controlTextDisabled,
								]}
							>
								+
							</Text>
						</TouchableOpacity>
						<View style={styles.controlCount}>
							<Text style={styles.controlCountText}>{selectedSamplesCount}</Text>
						</View>
						<TouchableOpacity style={styles.control} onPress={decrementSelectedSamples}>
							<Text
								style={[styles.controlText, selectedSamplesCount === 1 && styles.controlTextDisabled]}
							>
								-
							</Text>
						</TouchableOpacity>
					</View>
				</View>

				<View>
					<Text style={[styles.title, styles.inputTitle]}>Add Mailing Address</Text>
					<Input
						label='Full Name'
						value={fullName}
						onChange={setFullName}
						style={styles.input}
						error={fullNameError}
					/>
					<Input
						label='Address'
						value={address}
						onChange={setAddress}
						multiline
						numberOfLines={4}
						style={styles.input}
						error={addressError}
					/>
					<Input
						label='City'
						value={city}
						onChange={setCity}
						style={styles.input}
						error={cityError}
					/>
					<View style={styles.halfInputs}>
						<View style={styles.halfInputsContainer}>
							<Input
								label='State'
								value={state}
								onChange={setState}
								style={styles.input}
								error={stateError}
							/>
						</View>

						<View style={styles.halfInputsContainer}>
							<Input
								label='Zip Code'
								value={zipCode}
								onChange={setZipCode}
								style={styles.input}
								error={zipCodeError}
							/>
						</View>
					</View>
				</View>

				<View style={styles.total}>
					<Text style={styles.totalText}>Total</Text>
					<Text style={styles.totalText}>$ {selectedSamplesCount * 300}</Text>
				</View>

				<Button onClick={makeCheckout}>Checkout</Button>
			</KeyboardAvoidingView>
		</Screen>
	);
}
