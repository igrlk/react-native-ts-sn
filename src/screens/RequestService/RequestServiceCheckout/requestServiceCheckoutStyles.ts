import EStyleSheet from 'react-native-extended-stylesheet';
import { Dimensions } from 'react-native';

export default EStyleSheet.create({
	title: {
		fontFamily: 'Poppins',
		fontSize: 18,
		lineHeight: 27,
		color: '$colorBlack',
	},
	info: {
		marginTop: -4,
		display: 'flex',
		flexDirection: 'row',
	},
	infoText: {
		paddingLeft: 6,
		fontFamily: 'Poppins',
		fontSize: 12,
		lineHeight: 18,
		color: '#9E9E9E',
	},
	container: {
		marginTop: 22,
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'flex-start',
		marginBottom: 24,
	},
	controls: {
		flexDirection: 'row',
	},
	control: {
		width: 24,
		height: 30,
		borderWidth: 1,
		borderColor: '#E1E1E1',
		justifyContent: 'center',
		alignItems: 'center',
	},
	controlText: {
		color: '#9E9E9E',
		fontSize: 14,
		fontFamily: 'Poppins',
	},
	controlCount: {
		width: 40,
		height: 30,
		borderTopWidth: 1,
		borderBottomWidth: 1,
		borderColor: '#E1E1E1',
		justifyContent: 'center',
		alignItems: 'center',
	},
	controlCountText: {
		fontFamily: 'Poppins',
		fontSize: 16,
		color: '$colorBlack',
	},
	controlCountTextDisabled: {
		color: '#E1E1E1',
	},
	input: {
		marginBottom: 15,
	},
	inputTitle: {
		marginBottom: 10,
	},
	halfInputs: {
		flexDirection: 'row',
		justifyContent: 'space-between',
	},
	halfInputsContainer: {
		width: 140,
	},
	total: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		marginBottom: 15,
	},
	totalText: {
		fontFamily: 'Poppins',
		fontSize: 24,
		lineHeight: 36,
		color: '$colorBlack',
	},
});
