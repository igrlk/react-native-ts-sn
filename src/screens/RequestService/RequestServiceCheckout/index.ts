import RequestServiceCheckout from './RequestServiceCheckout';

export default RequestServiceCheckout;
