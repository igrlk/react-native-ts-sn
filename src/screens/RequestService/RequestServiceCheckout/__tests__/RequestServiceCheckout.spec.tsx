import React from 'react';
import renderer from 'react-test-renderer';

import RequestServiceCheckout from '../RequestServiceCheckout';

describe('RequestServiceCheckout', () => {
	it('match to snapshot', () => {
		const tree = renderer.create(<RequestServiceCheckout />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
