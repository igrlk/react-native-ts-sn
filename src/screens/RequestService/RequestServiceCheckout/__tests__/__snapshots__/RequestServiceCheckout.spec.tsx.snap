// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestServiceCheckout match to snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <View>
      <RCTScrollView>
        <View>
          <RCTScrollView
            automaticallyAdjustContentInsets={false}
            contentInset={
              Object {
                "bottom": 0,
              }
            }
            enableAutomaticScroll={true}
            enableOnAndroid={false}
            enableResetScrollToCoords={true}
            extraHeight={75}
            extraScrollHeight={0}
            getScrollResponder={[Function]}
            handleOnScroll={[Function]}
            keyboardDismissMode="interactive"
            keyboardOpeningTime={250}
            keyboardSpace={0}
            onScroll={[Function]}
            resetKeyboardSpace={[Function]}
            resetScrollToCoords={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            scrollEventThrottle={1}
            scrollForExtraHeightOnAndroid={[Function]}
            scrollIntoView={[Function]}
            scrollToEnd={[Function]}
            scrollToFocusedInput={[Function]}
            scrollToPosition={[Function]}
            showsVerticalScrollIndicator={true}
            update={[Function]}
            viewIsInsideTabBar={false}
          >
            <View>
              <View>
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight="22"
                    bbWidth="12"
                    fill="none"
                    focusable={false}
                    height="22"
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        undefined,
                        Object {
                          "opacity": 1,
                        },
                        Object {
                          "flex": 0,
                          "height": 22,
                          "width": 12,
                        },
                      ]
                    }
                    vbHeight={22}
                    vbWidth={12}
                    width="12"
                  >
                    <RNSVGGroup
                      fill={null}
                      fillOpacity={1}
                      fillRule={1}
                      font={Object {}}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    >
                      <RNSVGPath
                        d="M1.69385 10.9994L11.6294 1.06387C11.8738 0.819511 11.8738 0.427621 11.6294 0.183266C11.3851 -0.0610887 10.9932 -0.0610887 10.7488 0.183266L0.370644 10.5614C0.126289 10.8058 0.126289 11.1977 0.370644 11.442L10.7488 21.8156C10.8687 21.9355 11.0301 22 11.1868 22C11.3436 22 11.5049 21.9401 11.6248 21.8156C11.8692 21.5712 11.8692 21.1793 11.6248 20.935L1.69385 10.9994Z"
                        fill={
                          Array [
                            0,
                            4278190080,
                          ]
                        }
                        fillOpacity={1}
                        fillRule={1}
                        matrix={
                          Array [
                            1,
                            0,
                            0,
                            1,
                            0,
                            0,
                          ]
                        }
                        opacity={1}
                        originX={0}
                        originY={0}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                        rotation={0}
                        scaleX={1}
                        scaleY={1}
                        skewX={0}
                        skewY={0}
                        stroke={null}
                        strokeDasharray={null}
                        strokeDashoffset={null}
                        strokeLinecap={0}
                        strokeLinejoin={0}
                        strokeMiterlimit={4}
                        strokeOpacity={1}
                        strokeWidth={1}
                        vectorEffect={0}
                        x={0}
                        y={0}
                      />
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
                <Text>
                  Request Service
                </Text>
              </View>
              <View>
                <Text
                  style={
                    Array [
                      undefined,
                      undefined,
                    ]
                  }
                >
                  Add Mailing Address
                </Text>
                <View
                  style={Object {}}
                >
                  <View>
                    <Text
                      style={
                        Array [
                          undefined,
                          false,
                        ]
                      }
                    >
                      Full Name
                    </Text>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    onChangeText={[Function]}
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Array [
                        undefined,
                        undefined,
                        false,
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <View
                  style={Object {}}
                >
                  <View>
                    <Text
                      style={
                        Array [
                          undefined,
                          false,
                        ]
                      }
                    >
                      Address
                    </Text>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    multiline={true}
                    numberOfLines={4}
                    onChangeText={[Function]}
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Array [
                        undefined,
                        undefined,
                        false,
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <View
                  style={Object {}}
                >
                  <View>
                    <Text
                      style={
                        Array [
                          undefined,
                          false,
                        ]
                      }
                    >
                      City
                    </Text>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    onChangeText={[Function]}
                    rejectResponderTermination={true}
                    secureTextEntry={false}
                    style={
                      Array [
                        undefined,
                        undefined,
                        false,
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <View>
                  <View>
                    <View
                      style={Object {}}
                    >
                      <View>
                        <Text
                          style={
                            Array [
                              undefined,
                              false,
                            ]
                          }
                        >
                          State
                        </Text>
                      </View>
                      <TextInput
                        allowFontScaling={true}
                        autoCapitalize="none"
                        onChangeText={[Function]}
                        rejectResponderTermination={true}
                        secureTextEntry={false}
                        style={
                          Array [
                            undefined,
                            undefined,
                            false,
                          ]
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </View>
                  </View>
                  <View>
                    <View
                      style={Object {}}
                    >
                      <View>
                        <Text
                          style={
                            Array [
                              undefined,
                              false,
                            ]
                          }
                        >
                          Zip Code
                        </Text>
                      </View>
                      <TextInput
                        allowFontScaling={true}
                        autoCapitalize="none"
                        onChangeText={[Function]}
                        rejectResponderTermination={true}
                        secureTextEntry={false}
                        style={
                          Array [
                            undefined,
                            undefined,
                            false,
                          ]
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </View>
                  </View>
                </View>
              </View>
              <View>
                <View>
                  <Text>
                    Number of samples
                  </Text>
                  <View>
                    <View
                      accessible={true}
                      focusable={true}
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <RNSVGSvgView
                        align="xMidYMid"
                        bbHeight="17"
                        bbWidth="17"
                        fill="none"
                        focusable={false}
                        height="17"
                        meetOrSlice={0}
                        minX={0}
                        minY={0}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "borderWidth": 0,
                            },
                            undefined,
                            Object {
                              "opacity": 1,
                            },
                            Object {
                              "flex": 0,
                              "height": 17,
                              "width": 17,
                            },
                          ]
                        }
                        vbHeight={17}
                        vbWidth={17}
                        width="17"
                      >
                        <RNSVGGroup
                          fill={null}
                          fillOpacity={1}
                          fillRule={1}
                          font={Object {}}
                          matrix={
                            Array [
                              1,
                              0,
                              0,
                              1,
                              0,
                              0,
                            ]
                          }
                          opacity={1}
                          originX={0}
                          originY={0}
                          propList={
                            Array [
                              "fill",
                            ]
                          }
                          rotation={0}
                          scaleX={1}
                          scaleY={1}
                          skewX={0}
                          skewY={0}
                          stroke={null}
                          strokeDasharray={null}
                          strokeDashoffset={null}
                          strokeLinecap={0}
                          strokeLinejoin={0}
                          strokeMiterlimit={4}
                          strokeOpacity={1}
                          strokeWidth={1}
                          vectorEffect={0}
                          x={0}
                          y={0}
                        >
                          <RNSVGPath
                            d="M8.49964 0C3.80584 0 0 3.80584 0 8.49964C0 13.1934 3.80584 17 8.49964 17C13.1934 17 17 13.1934 17 8.49964C17 3.80584 13.1934 0 8.49964 0ZM10.2691 13.1733C9.83158 13.346 9.4833 13.477 9.2221 13.5676C8.96161 13.6583 8.65867 13.7036 8.31399 13.7036C7.78438 13.7036 7.37206 13.5741 7.07848 13.3158C6.78489 13.0574 6.63881 12.73 6.63881 12.3321C6.63881 12.1774 6.64961 12.0191 6.6712 11.8579C6.6935 11.6967 6.72876 11.5154 6.77697 11.3117L7.32457 9.37752C7.37278 9.19187 7.41452 9.01558 7.44762 8.85151C7.48072 8.68601 7.49655 8.53418 7.49655 8.39602C7.49655 8.14993 7.44546 7.97723 7.344 7.88008C7.2411 7.78294 7.04753 7.73545 6.75898 7.73545C6.61795 7.73545 6.47259 7.75632 6.32364 7.80021C6.17613 7.84554 6.04804 7.88656 5.94298 7.92686L6.08762 7.33105C6.44597 7.18497 6.78921 7.05977 7.11661 6.95615C7.44402 6.85109 7.75344 6.79928 8.04487 6.79928C8.57088 6.79928 8.97672 6.92737 9.26239 7.18066C9.54662 7.43467 9.68982 7.76495 9.68982 8.17079C9.68982 8.25498 9.67975 8.40322 9.66032 8.61477C9.64089 8.82705 9.60419 9.02061 9.55094 9.19835L9.00622 11.1268C8.96161 11.2815 8.92203 11.4585 8.88605 11.6564C8.85079 11.8543 8.83352 12.0054 8.83352 12.1069C8.83352 12.363 8.89037 12.5379 9.0055 12.6307C9.1192 12.7236 9.31852 12.7703 9.60059 12.7703C9.73371 12.7703 9.88267 12.7466 10.051 12.7005C10.218 12.6545 10.3389 12.6135 10.4152 12.5782L10.2691 13.1733ZM10.1727 5.34574C9.91865 5.58176 9.61283 5.69977 9.2552 5.69977C8.89829 5.69977 8.59031 5.58176 8.33414 5.34574C8.07941 5.10971 7.9506 4.8226 7.9506 4.48728C7.9506 4.15268 8.08013 3.86485 8.33414 3.62667C8.59031 3.38777 8.89829 3.26904 9.2552 3.26904C9.61283 3.26904 9.91936 3.38777 10.1727 3.62667C10.4267 3.86485 10.554 4.15268 10.554 4.48728C10.554 4.82332 10.4267 5.10971 10.1727 5.34574Z"
                            fill={
                              Array [
                                0,
                                4282234700,
                              ]
                            }
                            fillOpacity={1}
                            fillRule={1}
                            matrix={
                              Array [
                                1,
                                0,
                                0,
                                1,
                                0,
                                0,
                              ]
                            }
                            opacity={1}
                            originX={0}
                            originY={0}
                            propList={
                              Array [
                                "fill",
                              ]
                            }
                            rotation={0}
                            scaleX={1}
                            scaleY={1}
                            skewX={0}
                            skewY={0}
                            stroke={null}
                            strokeDasharray={null}
                            strokeDashoffset={null}
                            strokeLinecap={0}
                            strokeLinejoin={0}
                            strokeMiterlimit={4}
                            strokeOpacity={1}
                            strokeWidth={1}
                            vectorEffect={0}
                            x={0}
                            y={0}
                          />
                        </RNSVGGroup>
                      </RNSVGSvgView>
                    </View>
                    <Text>
                      $300/ sample
                    </Text>
                  </View>
                </View>
                <View>
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      -
                    </Text>
                  </View>
                  <View>
                    <Text>
                      1
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          undefined,
                          undefined,
                        ]
                      }
                    >
                      +
                    </Text>
                  </View>
                </View>
              </View>
              <View>
                <Text>
                  Total
                </Text>
                <Text>
                  $ 
                  300
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      undefined,
                      undefined,
                    ]
                  }
                >
                  Checkout
                </Text>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
